### PR TITLE
Display unapplied danger zone settings in settings page

### PR DESCRIPTION
**Pull request by a110605**
## Goal
- Add descriptions in danger zone to alert users some settings won't applied immediately ([document](https://longhorn.io/docs/1.7.0/references/settings/#danger-zone))
- Display unapplied settings

## Issue
https://github.com/longhorn/longhorn/issues/8071

## Result
after

### CODE REVIEW

File: src/routes/setting/setting.js [hunk 1]
```diff
@@ -132,17 +147,36 @@ const form = ({
...
   }).map(item => (
     <div key={item}>
       <div className={classnames(styles.fieldset, { [styles.dangerZone]: item === 'danger Zone' })}>
         <span className={styles.fieldsetLabel}>{item}</span>
+        {item === 'danger Zone' && (
+          <Alert
+            style={{ marginBottom: '1rem', marginTop: '1rem' }}
+            message={
+            <div className={styles.description}>
+              <span>
```
Comment (a110605) [inline src/routes/setting/setting.js:160]: @jillian-maroket , could you kindly help review or suggest these wordings?


Screenshot 2024-05-27 at 5 55 05 PM

File: src/routes/setting/setting.js [hunk 2]
```diff
@@ -132,17 +147,36 @@ const form = ({
...
+              Some danger zone settings won&apos;t applied immediately if there is any attached volume exists.
+              To make sure your changed settings are applied immediately, please detach all the volumes before saving the settings.<br />
+              For more information, please refer to <a target="blank" href="https://longhorn.io/docs/1.6.2/references/settings/#danger-zone">Longhorn danger zone document</a>
+              </span>
+              {unAppliedDangerZoneSettings.length > 0 && (
+                <>
+                  <br />
+                  <br />
+                  Below danger zone {unAppliedDangerZoneSettings.length === 1 ? 'setting has not' : 'settings have not'} not applied yet
```
Comment (jillian-maroket) [inline src/routes/setting/setting.js:169]:
```suggestion
                  The following {unAppliedDangerZoneSettings.length === 1 ? 'setting has not' : 'settings have not'} been applied:
```

**Review by jillian-maroket — COMMENTED** [review]
I have completed the review. 

The writing styles for UI strings and documentation are different. UI strings must be short but clear and complete in meaning. Links on the UI don't always need introductory words such as "For more information...".

**Comment by a110605** [comment]
> I have completed the review.
> 
> The writing styles for UI strings and documentation are different. UI strings must be short but clear and complete in meaning. Links on the UI don't always need introductory words such as "For more information...".

Thanks @jillian-maroket for review and feedback, I updated the description as below

Screenshot 2024-06-03 at 11 38 03 AM

**Review by jillian-maroket — APPROVED** [review]
LGTM. We need to discuss is how to best maintain doc links on the UI. The docs are versioned and the structure sometimes changes, resulting in broken links.

**Comment by a110605** [comment]
> LGTM. We need to discuss is how to best maintain doc links on the UI. The docs are versioned and the structure sometimes changes, resulting in broken links.

We can get longhorn version (v1.7.0 or v1.6.3 or v1.6.2...) in version file under root repo and append it in doc link.
Let me raise another PR to improve this part.

**Review by mantissahz — APPROVED** [review]

File: src/routes/setting/setting.js [hunk 3]
```diff
@@ -132,17 +147,34 @@ const form = ({
...
     <div key={item}>
       <div className={classnames(styles.fieldset, { [styles.dangerZone]: item === 'danger Zone' })}>
         <span className={styles.fieldsetLabel}>{item}</span>
+        {item === 'danger Zone' && (
+          <Alert
+            style={{ marginBottom: '1rem', marginTop: '1rem' }}
+            message={
+            <div className={styles.description}>
+              <span>
+              Some <a target="blank" href="https://longhorn.io/docs/1.6.2/references/settings/#danger-zone">Danger Zone settings</a> are not immediately applied when one or more volumes are still attached. Ensure that all volumes are detached before configuring the settings.
```
[the remark this answers is not in the document]
Comment (a110605) [reply src/routes/setting/setting.js:161]: Currently yes, to improve this part, either
 - simply change doc link to `https://longhorn.io/docs/`
 - implement mechanism in longhorn-ui to get current build version from vesion file. (I am trying this solution)

**Pull request merged**